src/main/java/gregtech/api/metatileentity/implementations/MTEMultiBlockBase.java

```diff
@@ -1038,62 +1039,86 @@ protected CheckRecipeResult doCheckRecipe() {
             return result;
         }
 
-        processingLogic.setInputFluids(getStoredFluids());
+        // Use hatch colors if any; fallback to color 1 otherwise.
+        short hatchColors = getHatchColors();
+        boolean doColorChecking = hatchColors != 0;
+        hatchColors = doColorChecking ? hatchColors : 1;
```

Comment:
minor nitpick: could you swap this to the following?
```suggestion
        if (!doColorChecking) hatchColors = 1;
```